builtin/logical/pki/cert_util.go

```diff
@@ -128,29 +170,66 @@ func validateCommonNames(req *logical.Request, commonNames []string, role *roleE
 			continue
 		}
 
-		if role.AllowTokenDisplayName {
-			if name == req.DisplayName {
+		if role.AllowLocalhost {
+			if name == "localhost" || (isEmail && emailDomain == "localhost") {
 				continue
 			}
 
 			if role.AllowSubdomains {
-				if strings.HasSuffix(name, "."+req.DisplayName) {
+				if strings.HasSuffix(sanitizedName, "."+req.DisplayName) {
+					// Regex won't match empty string, so protected against
+					// someone trying to get base without it being allowed
+					trimmed := strings.TrimSuffix(sanitizedName, "."+req.DisplayName)
+					if hostnameRegex.MatchString(trimmed) {
```

Comment:
This is nested under `AllowLocalhost` `AllowSubdomains` and has a suffix of ".DISPLAYNAME" and matches a hostname regexp? What is a possible input that even makes sense for this?